src/main/java/appeng/fluids/helper/DualityFluidInterface.java

```diff
@@ -113,18 +110,24 @@ public DualityFluidInterface( final AENetworkProxy networkProxy, final IFluidInt
 		this.fluids.setChangeSource( this.mySource );
 		this.items.setChangeSource( this.mySource );
 
-		this.tanks = new AEFluidTank[NUMBER_OF_TANKS];
 		this.requireWork = new IAEFluidStack[NUMBER_OF_TANKS];
-		this.configStacks = new IAEFluidStack[NUMBER_OF_TANKS];
-		this.tankChanged = new boolean[NUMBER_OF_TANKS];
 		for( int i = 0; i < NUMBER_OF_TANKS; ++i )
 		{
-			this.tanks[i] = new AEFluidTank( this, TANK_CAPACITY );
 			this.requireWork[i] = null;
-			this.configStacks[i] = null;
-			this.tankChanged[i] = false;
 		}
-		this.storage = new FluidHandlerConcatenate( this.tanks );
+	}
+
+	public IUpgradeableHost getHost()
+	{
+		if( this.iHost instanceof IUpgradeableHost )
+		{
+			return (IUpgradeableHost) this.iHost;
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unnecessary cast to "IUpgradeableHost". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1905)